proto-public/pbmesh/v1alpha1/upstreams_configuration.proto

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v1alpha1;
+
+import "google/protobuf/duration.proto";
+import "pbcatalog/v1alpha1/protocol.proto";
+import "pbcatalog/v1alpha1/selector.proto";
+import "pbmesh/v1alpha1/connection.proto";
+import "pbmesh/v1alpha1/routing.proto";
+import "pbresource/resource.proto";
+
+message UpstreamsConfiguration {
```

Comment:
I came here to say "why is this plural in the interior?" after seeing it up in the first file of the commit, but I see here that it affects several upstreams of a workload so i guess the plural makes sense.